### PR TITLE
Add Apple Sign In Authentication code sample for addition in repo

### PR DESCRIPTION
**Pull request by marcabreracast**
This is a PR in order to integrate a code sample regarding authentication with Apple Sign-In and the Realm Swift SDK

### CODE REVIEW

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//  AppDelegate.swift
+//  apple-sign-in
+//
+//  Created by Mar Cabrera on 20/01/2023.
+//
+
+import UIKit
+import RealmSwift
+
+let app = App(id: "")
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift]:
```suggestion
let app = App(id: "<ATLAS_APP_ID>")
```

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift:15]: Extra whitespaces, limit to no more than one whitespace or empty line

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        // Use this method to select a configuration to create the new scene with.
+        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
+    }
+
+    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
+        // Called when the user discards a scene session.
+        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
+        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
+    }
+
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift]: Extra whitespaces

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            }
+
+            let decodedToken = String(decoding: identityToken, as: UTF8.self)
+
+            print(decodedToken)
+
+            // Note: If the user decides to hide their email on authentication, full name will return an empty string and email will be nil
+
+            // Retrieve AppleID token so I can send it to Realm and authenticate myself
+
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift]: Extra line

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    @IBOutlet weak var appleSignInButton: ASAuthorizationAppleIDButton!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupAppleSignInButton()
+    }
+
+    private func setupAppleSignInButton() {
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift:23]: Will be helpful to add a comment about what this is doing, I deduce this is adding apples's sign in button.

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        appleSignInButton.addTarget(self, action: #selector(handleAppleIdRequest), for: .touchUpInside)
+        appleSignInButton.cornerRadius = 10
+    }
+
+    @objc func handleAppleIdRequest() {
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift:28]: Also, some comments here about what is happening would be helpful

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        let authorizationController = ASAuthorizationController(authorizationRequests: [request])
+        authorizationController.delegate = self
+        authorizationController.performRequests()
+    }
+}
+
+extension LoginViewController: ASAuthorizationControllerDelegate {
+    // Delegate that handles successful Sign In
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        if let appleIDCredential = authorization.credential as? ASAuthorizationAppleIDCredential {
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift:43]: In which case ASAuthorizationCredential is not ASAuthorizationAppleIDCredential?, why appleIDCredential.identityToken can be nil? Maybe some comments can be helpful here

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            switch result {
+            case .failure(let error):
+                print("Realm Login failed: \(error.localizedDescription)")
+                
+            case .success(_):
+                DispatchQueue.main.async {
+                    print("Successful Login")
+                    self.performSegue(withIdentifier: "goToWelcomeViewController", sender: nil)
+                }
+
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift]: Extra line

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    private func realmSignIn(appleToken: String) {
+        let credentials = Credentials.apple(idToken: appleToken)
+
+        app.login(credentials: credentials) { (result) in
+            switch result {
+            case .failure(let error):
+                print("Realm Login failed: \(error.localizedDescription)")
+
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift]: extra line

File: AppleSignInAuthentication/README.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
+# Integrating Apple Sign-in with Realm using the Swift SDK: A Step-by-Step Guide
+
+This example contains a skeleton app to be used as a reference on how to implement Apple Sign in authentication with Realm using the [Realm Swift SDK](https://www.mongodb.com/docs/atlas/app-services/authentication/apple/#apple-id-authentication).
+In this project you will be able to see an example of the following use case
+* Use Apple Sign In implementation in order to authenticate users in your Realm database.
+	* Log screen view in UIKit.
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/README.md:6]: Are these subitems of the above point?

File: AppleSignInAuthentication/README.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+## Scope
+
+* Create an iOS app in Swift using Xcode.
+* Integrate RealmSwift framework for database management.
+* Implement Apple Sign-In authentication.
+* Store user information securely in Realm database.
+
+**Realm Details:**
+* RealmSwift v10.34.1
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/README.md]: You should be using a more recent version of the SDK

File: AppleSignInAuthentication/README.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+└──  other configuration files...    
+```
+
+## Scope
+
+* Create an iOS app in Swift using Xcode.
+* Integrate RealmSwift framework for database management.
+* Implement Apple Sign-In authentication.
+* Store user information securely in Realm database.
+
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/README.md:23]: you could add a link to apple documentation where  apple sign and AuthenticationServices API are documented, so user can have a better understanding of this feature.

**Review by dianaafanador3 — COMMENTED** [review]
Great work on the sample, some small changes:
* You can add the README to the project (without compiling it), so It is easier to follow the instructions on it and the code
* Also, I'll pull the latest test changes from main so we can run `swiftlint` over the code. You can do this locally installing SwiftLint `brew install swiftlint` and run it on the root of your project to get lint issues.
* Remove target and files associated to Unit tests and UI test if there are no test added.
* Use a stable version of RealmSwift SDK instead of pointing to master.
* Do not leave any signing info pointing to the realm account on the sample project.
* I'll add a link to your blog post

**Comment by marcabreracast** [comment]
Thank you for the review Diana, just some follow-up questions on my side regarding a couple of points: 

>  You can add the README to the project (without compiling it), so It is easier to follow the instructions on it and the code
Do you mean adding another README other than the one that I currently added and that follows our template? 

> I'll add a link to your blog post
Are you referring to a link inside the code? Or in the README?

I have also pushed the latest changes with the reviewed suggestions

File: AppleSignInAuthentication/README.md (hunk 10 above)
Comment (marcabreracast) [reply AppleSignInAuthentication/README.md:6]: They are contained as a section inside the main headline, yes

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 13]
```diff
@@ -72,15 +79,12 @@ extension LoginViewController: ASAuthorizationControllerDelegate {
             switch result {
             case .failure(let error):
                 print("Realm Login failed: \(error.localizedDescription)")
-                
-            case .success(_):
+            case .success(let realm):
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift]:
```suggestion
            case .success(let user):
```

File: AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift [hunk 14]
```diff
@@ -72,15 +79,12 @@ extension LoginViewController: ASAuthorizationControllerDelegate {
...
                 DispatchQueue.main.async {
-                    print("Successful Login")
+                    print("Successful Login for realm \(realm)")
```
Comment (dianaafanador3) [inline AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift]:
```suggestion
                    print("Successful Login for realm \(user.id)")
```

**Comment by dianaafanador3** [comment]
@marcabreracast by adding the README is just adding it as a file into the project so we can see the README in Xcode. You can add the file like any other file but then remove the target membership so it doesn't get compile.
A link to the blog in the README will be helpful, it gives another reference for the customer so they understand the sample.
Just a few comments, and I still see the following listing issues.
```
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift:33:1: warning: Line Length Violation: Line should be 120 characters or less; currently it has 157 characters (line_length)
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/AppDelegate.swift:37:1: warning: Trailing Newline Violation: Files should have a single trailing newline (trailing_newline)
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/SceneDelegate.swift:16:1: warning: Line Length Violation: Line should be 120 characters or less; currently it has 122 characters (line_length)
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/SceneDelegate.swift:18:1: warning: Line Length Violation: Line should be 120 characters or less; currently it has 147 characters (line_length)
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/SceneDelegate.swift:26:1: warning: Line Length Violation: Line should be 120 characters or less; currently it has 140 characters (line_length)
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/SceneDelegate.swift:50:1: warning: Trailing Newline Violation: Files should have a single trailing newline (trailing_newline)
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/SceneDelegate.swift:19:19: warning: Unused Optional Binding Violation: Prefer `!= nil` over `let _ =` (unused_optional_binding)
/Users/diana.perez/Documents/Realm/realm-swift-samples/AppleSignInAuthentication/apple-sign-in/apple-sign-in/LoginViewController.swift:61:1: warning: Line Length Violation: Line should be 120 characters or less; currently it has 147 characters (line_length)
Done linting! Found 8 violations, 0 serious in 15 files.
```

**Review by dianaafanador3 — APPROVED** [review]

**Comment by dianaafanador3** [comment]
When merging use Squash merge, so we don't have a bunch of commits in the repository history

**Comment by marcabreracast** [comment]
I pushed the latest changes with the suggested code change. As soon as you approve it I'll go ahead and squash merge 🙂

**Review by dianaafanador3 — APPROVED** [review]

**Comment by dianaafanador3** [comment]
Remember to "squash and merge" like I mentioned above

**Pull request merged**